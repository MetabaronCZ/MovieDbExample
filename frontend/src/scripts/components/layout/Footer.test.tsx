import { describe, expect, it } from '@jest/globals';
import { getByText } from '@testing-library/dom';

import { Footer } from './Footer';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/layout/Footer', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(<Footer />, {
      wrapper: TestComponentWrapper,
    });
    expect(container).toMatchSnapshot();

    const footer = container.getElementsByTagName('footer');
    expect(footer.length).toEqual(1);

    const year = new Date().getFullYear();

    const copyright = getByText(container, `Â© ${year}`, { exact: false });
    expect(copyright).toBeInTheDocument();
  });
});
