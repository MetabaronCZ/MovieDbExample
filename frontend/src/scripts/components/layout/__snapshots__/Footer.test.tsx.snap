// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/layout/Footer should render properly 1`] = `
.c1 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  color: inherit;
  text-decoration: underline;
}

.c1:hover {
  text-decoration: none;
}

.c1:focus {
  outline: 1px dashed #002d56;
  outline-color: inherit;
}

.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 12px;
  line-height: 24px;
  padding: 0 16px;
  color: #98c3d9;
  text-align: right;
}

<div>
  <footer
    class="c0"
  >
    Copyright Â© 2025 
    <a
      class="c1"
      href="https://github.com/MetabaronCZ"
      target="_blank"
    >
      https://github.com/MetabaronCZ
    </a>
  </footer>
</div>
`;
